lldb-eval/context.cc

```diff
@@ -271,27 +297,16 @@ lldb::SBValue Context::LookupIdentifier(const std::string& name) const {
     // resolved relative to the current scope. I.e. if the current frame is in
     // "ns1::ns2::Foo()", then "ns2::x" should resolve to "ns1::ns2::x".
 
-    // List global variable with the same "basename". There can be many matches
-    // from other scopes (namespaces, classes), so we do additional filtering
-    // later.
-    lldb::SBValueList values = ctx_.GetTarget().FindGlobalVariables(
-        name_ref.data(), /*max_matches=*/std::numeric_limits<uint32_t>::max());
-
-    // Find the corrent variable by matching the name. lldb::SBValue::GetName()
-    // can return strings like "::globarVar", "ns::i" or "int const ns::foo"
-    // depending on the version and the platform.
-    for (uint32_t i = 0; i < values.GetSize(); ++i) {
-      lldb::SBValue val = values.GetValueAtIndex(i);
-      llvm::StringRef val_name = val.GetName();
-
-      if (val_name == name_ref ||
-          val_name == llvm::formatv("::{0}", name_ref).str() ||
-          val_name.endswith(llvm::formatv(" {0}", name_ref).str()) ||
-          val_name.endswith(llvm::formatv("*{0}", name_ref).str()) ||
-          val_name.endswith(llvm::formatv("&{0}", name_ref).str())) {
-        value = val;
-        break;
-      }
+    if (scope_ && !name_ref.contains("::")) {
```

Comment:
Should it be possible to lookup static variables from nested types?

```c++
struct Outer {
  struct Inner {
    static constexpr int x = 1;
  };
} foo;

// eval(scope="foo", expr="Inner::x + 1") 
```

Reply:
Good point! Replaced it with `!global_scope` and added more tests.